pkg/codegen/docs/gen.go

```diff
@@ -809,6 +834,11 @@ func (mod *modContext) getConstructorResourceInfo(resourceTypeName string) map[s
 		case "nodejs", "go":
 			// Intentionally left blank.
 		case "csharp":
+			if mod.mod == "" {
```

Comment:
This actually fixes a bug for _all_ providers right now. The `Provider` resource will never have a module name and as such its link is always at the package-level.